qiskit/pulse/reschedule.py

```diff
@@ -141,27 +141,35 @@ def add_implicit_acquires(schedule: ScheduleComponent, meas_map: List[List[int]]
     """
     new_schedule = Schedule(name=schedule.name)
 
+    # Dict[qubit, Dict[time, AcquireInstruction]]
+    acquires_map = dict()
     for time, inst in schedule.instructions:
         if isinstance(inst, AcquireInstruction):
             if any([acq.index != mem.index for acq, mem in zip(inst.acquires, inst.mem_slots)]):
                 warnings.warn("One of your acquires was mapped to a memory slot which didn't match"
                               " the qubit index. I'm relabeling them to match.")
-            cmd = Acquire(inst.duration, inst.command.discriminator, inst.command.kernel)
-            # Get the label of all qubits that are measured with the qubit(s) in this instruction
-            existing_qubits = {chan.index for chan in inst.acquires}
-            all_qubits = []
-            for sublist in meas_map:
-                if existing_qubits.intersection(set(sublist)):
-                    all_qubits.extend(sublist)
-            # Replace the old acquire instruction by a new one explicitly acquiring all qubits in
-            # the measurement group.
-            new_schedule |= AcquireInstruction(
-                cmd,
-                [AcquireChannel(i) for i in all_qubits],
-                [MemorySlot(i) for i in all_qubits]) << time
+
+            for acq in inst.acquires:
+                if acq.index not in acquires_map:
+                    acquires_map[acq.index] = {time: inst}
```

Comment:
I think this is a test you should add:

```
def test_multiple_acquires(self):
    """<description>"""
    acq_q0 = Acquire(1200)(AcquireChannel(0), MemorySlot(0))
    sched += acq_q0
    sched += acq_q0 << sched.duration
    sched  = add_implicit_acquires(add_implicit_acquires, meas_map=[[0]])
    self.assertEqual(sched.instructions, ((0, acq_q0), (1200, acq_q0)))
```
It appears that the second acquire might be dropped

Reply:
added